lib/jsonwp-proxy/proxy.js

```diff
@@ -232,15 +236,111 @@ class JWProxy {
     }
   }
 
+  async getDownstreamProtocol () {
+    if (!this._downstreamProtocol) {
+      const [, resBody] = await this.proxy('/status', 'GET');
+      this._downstreamProtocol = this.getProtocolFromResBody(resBody);
+    }
+    return this._downstreamProtocol;
+  }
+
+  requestToCommandName (url, method) {
+    const extractCommandName = (pattern) => {
+      const pathMatch = pattern.exec(url);
+      return pathMatch ? routeToCommandName(pathMatch[1], method) : null;
+    };
+    let commandName = routeToCommandName(url, method);
+    if (!commandName && _.includes(url, '/wd/hub/session/')) {
+      commandName = extractCommandName(/\/wd\/hub\/session\/[^\/]+(.+)/);
+    }
+    if (!commandName && _.includes(url, '/wd/hub/')) {
+      commandName = extractCommandName(/\/wd\/hub(\/.+)/);
+    }
+    return commandName;
+  }
+
+  async proxyCommand (url, method, body = null) {
+    const commandName = this.requestToCommandName(url, method);
+    if (!commandName) {
+      return await this.proxy(url, method, body);
+    }
+    log.debug(`Matched '${url}' to command name '${commandName}'`);
+
+    // Handle "crossing" endpoints for the case
+    // when upstream and downstream drivers operate different protocols
+
+    // Same url, but different arguments
+
+    if (commandName === 'timeouts')  {
+      return await this.proxySetTimeouts(url, method, body);
+    }
+
+    // Same arguments, but different URLs
+
+    const conflicts = [
```

Comment:
i don't care much one way or another, but this could be a file-level constant (and the embedded functions would then just take a url param)

Reply:
yes, it is easy to do that.